Formula/asciiquarium.rb

```diff
@@ -37,6 +39,39 @@ def install
     resources.each do |r|
       r.stage do
         system "perl", "Makefile.PL", "INSTALL_BASE=#{libexec}"
+        if (r.name == "Curses") && (MacOS.version >= :big_sur)
+          # Work around a 11.0 problem in MakeMaker:
+          #   https://github.com/Perl-Toolchain-Gang/ExtUtils-MakeMaker/issues/381
+          #
+          # This issue is that due to the changes to how libraries
+          # exist (or, rather, DON'T exist) on the filesystem, usual ways
+          # of detecting libraries do not work.  This extends even
+          # to the MakeMaker that comes bundled with Apple's own
+          # copy of perl (at least as of OS 11.0.1)
+          #
+          # The result is that Makefile.PL generates a Makefile that
+          # does not contain the expected "LDLOADLIBS = -lcurses" line.
+          # The tell is when it prints the diagnostic:
+          #   Warning (mostly harmless): No library found for -lcurses
+          # Since it doesn't find the library, it assumes no flag is needed.
+          #
+          # Normally we could probably work around this by just specifying
+          # an explicit value of LDLOADLIBS on the "make" command line,
+          # but that doesn't work here.  That is because the Makefile
+          # runs the perl script "test.syms" which then opens and re-parses
+          # "Makefile" to find these values and then uses them to introspect
+          # the system's curses library.  If it doesn't find the right
+          # value of LDLOADLIBS *inside the Makefile* it won't find any
+          # symbols and the build will end up not working.
+          #
+          # Therefore we need to actually munge "Makefile" after "Makefile.PL"
+          # runs but before "make" does
```

Comment:
```suggestion
          # Work around a 11.0 problem in MakeMaker:
          #   https://github.com/Perl-Toolchain-Gang/ExtUtils-MakeMaker/issues/381
```

Reply:
I know the comment block is long, but it provides a lot of context that took me a *long* time to figure out.  Just referencing that MakeMaker bug does nothing to explain the intricacies of why we have to munge `Makefile` for `test.syms`'s benefit which is very non-obvious.

Reply:
@mitchblank, perhaps you'd like to document this in a gist that you can link to in this comment block? Maybe that will achieve a compromise between keeping the formula clean but also having the necessary information easy to access.